### PR TITLE
fix(coverage): ignore urls from doc testing

### LINKED ISSUE #25732: bug: `deno coverage` unexpectedly failing on simple snippet
See https://github.com/denoland/std/actions/runs/10941340431/job/30375761825?pr=6018

```
Task cov:gen deno task cov --lcov --output=cov.lcov
Task cov deno coverage --ignore="**/*.generated.mjs,**/_test_utils.ts,.github" "--lcov" "--output=cov.lcov"
error: Failed to fetch "file:///home/runner/work/std/std/assert/equal.ts$23-29.ts" from cache.
          Before generating coverage report, run `deno test --coverage` to ensure consistent state.
Error: Process completed with exit code 1.
```

Failing on this simple snippet:
https://github.com/denoland/std/blob/6a4eb6cb9185b0b9e0816a227c04b5bf039b99ef/assert/equals.ts#L19-L24

To reproduce:
1. Clone [`std`](https://github.com/denoland/std)
2. Checkout https://github.com/denoland/std/pull/6018
3. Run `deno task test`
4. Run `deno task cov:gen`

CC @magurotuna

### PR DESCRIPTION
**Pull request by kt3k**
We started executing doc examples in #25220. Because of that change, now the coverage data for generated urls like `file:///path/to/module.ts$20-30.ts` are generated, and `deno coverage` fails by trying to fetch that file from the cache. This behavior causes #25732

This PR fixes the above by ignoring urls of the pattern `\$\d+-\d+\.(ts|js)$` when collecting the coverage data.

closes #25732

### CODE REVIEW

File: cli/tools/coverage/mod.rs [hunk 1]
```diff
@@ -452,6 +452,10 @@ fn filter_coverages(
   let exclude: Vec<Regex> =
     exclude.iter().map(|e| Regex::new(e).unwrap()).collect();
 
+  // Matches virtual file paths for doc testing
+  // e.g. file:///path/to/mod.ts$23-29.ts
+  let doc_test_re = Regex::new(r"\$\d+-\d+\.(ts|js)$").unwrap();
```
Comment (kt3k) [inline cli/tools/coverage/mod.rs:457]: @magurotuna Do you know what are the possible file extensions from doc example extraction?

Comment (magurotuna) [reply cli/tools/coverage/mod.rs:457]: The extension is taken from the code fence, like:

````markdown
```ts
// becomes .ts
```

```tsx
// becomes .tsx
```
````

Possible extensions are listed here
https://deno-docs--magurotuna-improved-doc-te.deno.dev/runtime/fundamentals/testing/#documentation-tests

- js
- javascript
- mjs
- cjs
- jsx
- ts
- typescript
- mts
- cts
- tsx

Comment (kt3k) [reply cli/tools/coverage/mod.rs:457]: Thanks!

**Review by magurotuna — APPROVED** [review]
LGTM

**Pull request merged**